diesel/src/sqlite/mod.rs

```diff
@@ -12,5 +12,19 @@ pub mod query_builder;
 
 pub use self::backend::{Sqlite, SqliteType};
 pub use self::connection::SqliteConnection;
-pub use self::connection::raw::Aggregator;
 pub use self::query_builder::SqliteQueryBuilder;
+
+/// The implementation details for a SQLite aggregate function
+///
+/// This trait is to be used in conjunction with the `sql_function!`
+/// macro for defining a custom SQLite aggregate function.
+pub trait SqliteAggregateFunction<Args>: Default {
+    /// The result type of the SQLite aggregate function
+    type Output;
+
+    /// The `step()` method is called once for every record of the query
+    fn step(&mut self, args: Args);
+    /// After the last row has been processed, the `finalize()` method is
+    /// called to compute the result of the aggregate function
+    fn finalize(&self) -> Self::Output;
```

Comment:
As we now move out the aggregator context in the finalize function we can just do 
```suggestion
    fn finalize(self) -> Self::Output;
```
here, right?
This should lower the overhead for cases where we would return a `String` for example, as we don't need to clone it anymore.